src/mail/InboxRuleHandler.js

```diff
@@ -125,7 +125,9 @@ function _checkContainsRule(value: string, inboxRule: InboxRule): ?InboxRule {
 /** export for test. */
 export function _matchesRegularExpression(value: string, inboxRule: InboxRule): boolean {
 	if (isRegularExpression(inboxRule.value)) {
-		let regExp = new RegExp(inboxRule.value.substring(1, inboxRule.value.length - 1));
+		let flags = inboxRule.replace(/.*\/([gimsuy]*)$/, '$1');
+		let pattern = inboxRule.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
```

Comment:
Sorry, I totally missed this on the first review.

```suggestion
		let flags = inboxRule.value.replace(/.*\/([gimsuy]*)$/, '$1');
		let pattern = inboxRule.value.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
```

Reply:
Ah, yes. Copy pasta from fiddle.